src/helm/clients/vertexai_client.py

```diff
@@ -194,12 +201,22 @@ def get_model(model_name: str) -> GenerativeModel:
 
     def make_request(self, request: Request) -> RequestResult:
         """Make a request"""
-        contents: str = request.prompt
+        contents = [request.prompt]
+        content_key = request.prompt
 
         # For the multimodal case, build up the content with the media objects of `request.multimodal_prompt`
         if request.multimodal_prompt is not None:
             return self._make_multimodal_request(request)
 
+        if request.messages is not None:
+            contents = []
+            role_mapping = {"user": "user", "assistant": "model"}
+            for msg in request.messages:
+                contents.append(
+                    Content(role=role_mapping.get(msg["role"], "user"), parts=[Part.from_text(msg["content"])])
+                )
+            content_key = "\n".join([msg["content"] for msg in request.messages])
```

Comment:
don't need this - can just rely on the existing `content` field (the cache key can be a nested dict).

Reply:
Just to confirm I understand it correctly, do you mean we can use the `Content` objects as the cache key? But here we have a list of messages, which essentially gives a list of `Content` objects, does a list of such nested dicts also work? If it does I'm curious about how that works - any quick pointers would be much appreciated.

Reply:
Yes, basically any nested dict that can be serialized to JSON can be used as a cache key. In terms of the implementation:

1. MongoDB natively supports using a JSON object 
2. Other caches like SQLite serialize the JSON object to a string in a "canonical" way, which happens here:

https://github.com/stanford-crfm/helm/blob/40b3d23ccc1b62cae48cee6fe7665d75cfdb0566/src/helm/common/key_value_store.py#L9-L11

In terms of this PR, you could do something like

```
if:
    cache_key["messages"] = request.messages
else:
    cache_key["prompt"] = request.prompt
```

or

```
{
    "prompt": request.messages or request.prompt
}
```

or something similar